### PR TITLE
infosync, helper: add caller ID to the PD HTTP client on the hot path

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #35319, https://github.com/tikv/pd/issues/7300.

Problem Summary:

Add caller ID to the PD HTTP client on the hot path.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/domain/infosync/info.go [hunk 1]
```diff
@@ -200,7 +200,9 @@ func GlobalInfoSyncerInit(
 	skipRegisterToDashBoard bool,
 ) (*InfoSyncer, error) {
 	if pdHTTPCli != nil {
-		pdHTTPCli = pdHTTPCli.WithRespHandler(pdResponseHandler)
+		pdHTTPCli = pdHTTPCli.
+			WithCallerID("tidb-info-syncer").
```
Comment (zimulala) [inline pkg/domain/infosync/info.go:204]: Do we need to add a constant to this `CallerID`?

File: pkg/store/helper/helper.go [hunk 2]
```diff
@@ -93,11 +97,15 @@ func NewHelper(store Storage) *Helper {
...
 func (h *Helper) TryGetPDHTTPClient() (pd.Client, error) {
+	if h.pdHTTPCli != nil {
+		return h.pdHTTPCli, nil
+	}
 	cli := h.Store.GetPDHTTPClient()
 	if cli == nil {
 		return nil, errors.New("pd http client unavailable")
 	}
-	return cli, nil
+	h.pdHTTPCli = cli.WithCallerID("tidb-store-helper")
```
Comment (zimulala) [inline pkg/store/helper/helper.go:107]: Ditto.

File: pkg/domain/infosync/info.go (hunk 1 above)
Comment (JmPotato) [reply pkg/domain/infosync/info.go:204]: It seems that currently only this place will use this string. I feel like const is not that necessary?

**Review by zimulala — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**